src/NLog.Web.AspNetCore/NLogRequestPostedBodyMiddlewareConfiguration.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// Contains the configuration for the NLogRequestPostedBodyMiddleware
+    /// </summary>
+    public class NLogRequestPostedBodyMiddlewareConfiguration
+    {
+        /// <summary>
+        /// The default configuration
+        /// </summary>
+        public static readonly NLogRequestPostedBodyMiddlewareConfiguration Default = new NLogRequestPostedBodyMiddlewareConfiguration();
+
+        /// <summary>
+        /// Defaults to true
+        /// </summary>
+        public bool DetectEncodingFromByteOrderMark { get; set; } = true;
+
+        /// <summary>
+        /// The maximum request size that will be captured
+        /// Defaults to 30KB
+        /// </summary>
+        public int MaximumRequestSize { get; set; } = 30 * 1024;
```

Comment:
Can it be renamed to `MaxPostedBodySize` ? Maybe update comments about how it checks against `ContentLength` to protect against "very large" posted-body-payloads. And that EnableBuffering has an upper-limit of 30KByte where it changes to TEMP-files.